### PR TITLE
chore: [FX-4015] Integrate Rich Text Editor to Forms

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-4015]

### Description

We need to integrate a new RTE to Picasso Forms.

### How to test

- Check the regress on temploy

### Screenshots

Picasso | Form 🔊 2023-06-13 13-09-49

### Development checks

- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Ensure that deployed demo has expected results and good examples
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4015]: https://toptal-core.atlassian.net/browse/FX-4015?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso/src/LexicalEditor/LexicalEditor.tsx [hunk 1]
```diff
@@ -150,9 +151,13 @@ const LexicalEditor = forwardRef<HTMLDivElement, Props>(function LexicalEditor(
   const handleChange = useCallback(
     (editorState, editor) => {
       editorState.read(() => {
-        const htmlValue = $generateHtmlFromNodes(editor, null)
+        const isEmpty = $isRootTextContentEmpty(editor.isComposing(), false)
```
Comment (dmaklygin) [inline packages/picasso/src/LexicalEditor/LexicalEditor.tsx:154]: to check the emptyness of the content

File: packages/picasso/src/RichTextEditor/RichTextEditor.tsx [hunk 2]
```diff
@@ -175,13 +176,25 @@ export const RichTextEditor = forwardRef<HTMLDivElement, Props>(
...
+const hiddenInputStyle: React.CSSProperties = {
+  position: 'absolute',
+  opacity: 0,
+  zIndex: -1,
+}
+
+// Native `for` attribute on label does not work for div target
+const enableFocusOnLabelClick = (hiddenInputId: string) => (
+  <input type='text' id={hiddenInputId} style={hiddenInputStyle} />
+)
```
Comment (augustobmoura) [inline packages/picasso/src/RichTextEditor/RichTextEditor.tsx:196]: Any reason to split this on a function? Maybe we can inline it on line 179 just for consistency, wdyt?

File: packages/picasso/src/RichTextEditor/RichTextEditor.tsx
Comment (TomasSlama) [inline packages/picasso/src/RichTextEditor/RichTextEditor.tsx:1]: Should we also take care of the `highlight` funcitontality that we have in forms?

Comment (dmaklygin) [reply packages/picasso/src/RichTextEditor/RichTextEditor.tsx:1]: I'll take a look whether it depends. thanks

Comment (dmaklygin) [reply packages/picasso/src/RichTextEditor/RichTextEditor.tsx:1]: done

**Review by TomasSlama — APPROVED** [review]

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

File: packages/picasso/src/RichTextEditor/test.tsx [hunk 3]
```diff
@@ -181,4 +181,24 @@ describe('RichTextEditor', () => {
       })
     })
   })
+
+  describe('when status prop is "default"', () => {
```
Comment (sashuk) [inline packages/picasso/src/RichTextEditor/test.tsx:185]: Should there be a test for highlighting? Or it is covered by a visual test?

**Review by sashuk — APPROVED** [review]
Tested on temploy; error and highlighted states work as expected!

File: packages/picasso/src/RichTextEditor/test.tsx (hunk 3 above)
Comment (dmaklygin) [reply packages/picasso/src/RichTextEditor/test.tsx:185]: it should be better to catch with visual tests. we have them

**Review by augustobmoura — APPROVED** [review]

**Pull request merged**